assets/app/scripts/controllers/attachPersistentVolumeClaim.js

```diff
@@ -0,0 +1,280 @@
+          $scope.disableInputs = true;
+          $scope.alerts['attach-persistent-volume-claim'] = {
+            type: "error",
+            message: errorMessage,
+            details: errorDetails
+          };
+        };
+
+        // check parameters validity
+        var validate = function() {
+          if ($scope.deploymentConfigName && $scope.deploymentName) {
+            sendError("Deployment and deployment config can't be both provided provided.");
+            return false;
+          }
+          if (!$scope.deploymentConfigName && !$scope.deploymentName) {
+            sendError("A deployment or deployment config must be provided.");
+            return false;
+          }
+          return true;
+        };
+
+        // load resources required to show the page (list of pvcs and deployment or deployment config)
+        var load = function() {
+          if ($scope.deploymentConfigName) {
+            DataService.get("deploymentconfigs", $scope.deploymentConfigName, context).then(
+              function(deploymentConfig) {
+                angular.forEach(deploymentConfig.spec.template.spec.containers, function(container) {
+                  $scope.attach.containers.individual[container.name] = true;
+                });
+                $scope.attach.deploymentConfig = deploymentConfig;
+                rebuildBreadcrumb();
+              },
+              function(e) {
+                sendError("The deployment config could not be loaded.", getErrorDetails(e));
+              }
+            );
+          }
+
+          if ($scope.deploymentName) {
+            DataService.get("replicationcontrollers", $scope.deploymentName, context).then(
+              function(deployment) {
+                angular.forEach(deployment.spec.template.spec.containers, function(container) {
+                  $scope.attach.containers.individual[container.name] = true;
+                });
+                $scope.attach.deployment = deployment;
+                rebuildBreadcrumb();
+              },
+              function(e) {
+                sendError("The deployment could not be loaded.", getErrorDetails(e));
+              }
+            );
+          }
+
+          DataService.list("persistentvolumeclaims", context, 
+            function(pvcs) {
+              $scope.pvcs = orderByDisplayName(pvcs.by("metadata.name"));
+              if (!$scope.pvcs.length) {
+                sendError("There are no persistent volumes in your project to attach.");
+              } else {
+                if (!$scope.attach.persistentVolumeClaim) {
+                  $scope.attach.persistentVolumeClaim = $scope.pvcs[0];
+                }
+              }
+            },
+            function(e) {
+              sendError("The persistent volume claims could not be loaded.", getErrorDetails(e));
+            }
+          );
+
+          return true;
+        };
+
+        var volumeNameIsUnique = function(name, podTemplate) {
```

Comment:
Maybe it's my java background, but I like function names that start with `is`. Then it's obvious they're checking something and returning a boolean with no side effects.